### PR TITLE
GH-110796: fix intermittent test failure in test_current_exceptions

### PR DESCRIPTION
**Pull request by FFY00**
* Issue: gh-110796

### CODE REVIEW

**Comment by FFY00** [comment]
@jd, I know it has been a while, but as the original author, could you have a look? Did I missing something, or is my interpretation sane? Thanks!

File: Lib/test/test_sys.py [hunk 1]
```diff
@@ -520,17 +520,17 @@ def f123():
...
                     raise ValueError("oops")
                 except ValueError:
+                    g_raised.set()
                     if leave_g.wait(timeout=support.LONG_TIMEOUT):
                         break
 
         t = threading.Thread(target=f123)
         t.start()
-        entered_g.wait()
+        g_raised.wait(timeout=support.LONG_TIMEOUT)
```
Comment (FFY00) [inline Lib/test/test_sys.py:533]: Also added a timeout here, as if an unexpected exception is raised in the thread, the event will never be set, and we will deadlock.

**Comment by FFY00** [comment]
The Windows failure is GH-110800.

**Review by jd — CHANGES_REQUESTED** [review]
This looks like a good fix, you need to change the comment on L536 though 😄

**Comment by FFY00** [comment]
Thanks! I forgot that.

**Pull request merged**